src/Bicep.Core/Registry/Oci/ArtifactIdParts.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Bicep.Core.Registry.Oci
+{
+    public class ArtifactIdParts : IArtifactIdParts
```

Comment:
Should we implement `IEqualtable<ArtifactIdParts>`?

Reply:
I looked into that.  None of them currently implement IEquatable, and as far as I can tell, it's just a minor performance optimization.  I didn't want to risk messing anything else up so I'm leaving as is for now.